File: noise/src/main/java/com/paramsen/noise/NoiseNativeBridge.java
package com.paramsen.noise;

/**
 * JNI interface
 *
 * @author Pär Amsen 06/2017
 */
public class NoiseNativeBridge {
    static {
        System.loadLibrary("noise");
    }

    public static native void realThreadSafe(float[] in, float[] out);
    public static native long realOptimizedCfg(int inSize);
    public static native void realOptimized(float[] in, float[] out, long cfgPointer);
    public static native long realOptimizedCfgDispose(long cfgPointer);

    public static native void imaginaryThreadSafe(float[] in, float[] out);
    public static native long imaginaryOptimizedCfg(int inSize);
    public static native void imaginaryOptimized(float[] in, float[] out, long cfgPointer);
    public static native long imaginaryOptimizedCfgDispose(long cfgPointer);
}

File: noise/src/main/java/com/paramsen/noise/NoiseNativeBridge.kt
package com.paramsen.noise

/**
 * JNI interface
 *
 * @author Pär Amsen 06/2017
 */
object NoiseNativeBridge {
    init {
        System.loadLibrary("noise")
    }

    external fun realConfig(inputLength: Int): Long
    external fun realConfigDispose(cfgPointer: Long): Long
    external fun real(`in`: FloatArray, out: FloatArray, cfgPointer: Long)

    external fun imaginaryConfig(inputLength: Int): Long
    external fun imaginaryConfigDispose(cfgPointer: Long): Long
    external fun imaginary(`in`: FloatArray, out: FloatArray, cfgPointer: Long)
}
